internal/controller/siteconfig_builder.go

```diff
@@ -107,48 +107,48 @@ func (scbuilder *SiteConfigBuilder) renderTemplates(ctx context.Context, c clien
 		}
 
 		// process Template ConfigMap
-		for kind, template := range templatesConfigMap.Data {
-			if suppressRenderingManifest(kind, suppressedManifests) {
-				scbuilder.Log.Info(fmt.Sprintf("renderTemplates: skipping rendering template manifest %s for SiteConfig %s", kind, siteConfig.Name))
-				continue
-			}
+		for templateKey, template := range templatesConfigMap.Data {
 
 			siteData, err := buildSiteData(siteConfig, node)
 			if err != nil {
 				scbuilder.Log.Error(err, fmt.Sprintf("renderTemplates: failed to build SiteConfig data for SiteConfig %s", siteConfig.Name))
 				return nil, err
 			}
 
-			manifest, err := scbuilder.render(kind, template, siteData)
+			manifest, err := scbuilder.render(templateKey, template, siteData)
 			if err != nil {
 				scbuilder.Log.Error(err, fmt.Sprintf("renderTemplates: failed to render templateRef %s for SiteConfig %s", templateRef.Name, siteConfig.Name))
 				return manifests, err
 			}
 
 			if manifest != nil {
 				kind := manifest["kind"].(string)
-				if node == nil {
-					// Append cluster-level user provided extra annotations if exist
-					if extraManifestAnnotations, ok := siteConfig.Spec.ExtraAnnotationSearch(kind); ok {
-						manifest = appendManifestAnnotations(extraManifestAnnotations, manifest)
+				if !suppressManifest(kind, suppressedManifests) {
+					if node == nil {
+						// Append cluster-level user provided extra annotations if exist
+						if extraManifestAnnotations, ok := siteConfig.Spec.ExtraAnnotationSearch(kind); ok {
+							manifest = appendManifestAnnotations(extraManifestAnnotations, manifest)
+						}
+					} else {
+						// Append node-level user provided extra annotations if exist
+						if extraCRAnnotations, ok := node.ExtraAnnotationSearch(kind, &siteConfig.Spec); ok {
+							manifest = appendManifestAnnotations(extraCRAnnotations, manifest)
+						}
 					}
+					manifests = append(manifests, manifest)
 				} else {
-					// Append node-level user provided extra annotations if exist
-					if extraCRAnnotations, ok := node.ExtraAnnotationSearch(kind, &siteConfig.Spec); ok {
-						manifest = appendManifestAnnotations(extraCRAnnotations, manifest)
-					}
+					scbuilder.Log.Info(fmt.Sprintf("renderTemplates: suppressing manifest %s for SiteConfig %s", kind, siteConfig.Name))
```

Comment:
I prefer the guard clause to using `if .. else ..`
It leads to less of a chance of errors later when someone adds something after the else and less indentation which makes things generally easier to read.

Reply:
i.e.

```go
if suppressManifest(...) {
  scbuilder.Log.Info(...)
  continue
}
if node == nil {
...
```